components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3568,7 +3577,11 @@ where
     EK: KvEngine,
 {
     fn drop(&mut self) {
-        self.delegate.clear_all_commands_as_stale();
+        if tikv_util::thread_group::is_shutdown(!cfg!(test)) {
+            self.delegate.clear_all_commands_silently()
```

Comment:
Looks like a no-op, the callback should be droped while `ApplyFsm` is dropped.

Reply:
We have assert to ensure every pending command is called properly to avoid latch leak.